atlas/responder/pkg/svc/zserver.go

```diff
@@ -0,0 +1,341 @@
+		if err := a.PublishMsg(ctx, id, "getRequests", nil); err != nil {
+			return nil, err
+		}
+		time.Sleep(time.Millisecond * 15)
+		result, ok := a.responses.Load(id)
+		if !ok {
+			return nil, errors.New(errorMissingResp)
+		}
+		a.responses.Delete(id)
+		i, err := strconv.Atoi(result.(map[string]string)["Value"])
+		if err != nil {
+			return nil, err
+		}
+		return &pb.GetRequestsResponse{Value: int32(i)}, nil
+	}
+
+	return nil, errors.New(invalidServiceName)
+}
+
+func (a *Responder) Reset(ctx context.Context, in *pb.ResetRequest) (*pb.ResetResponse, error) {
+	if in == nil || in.Service == "" {
+		return nil, errors.New(emptyRequest)
+	}
+
+	if in.Service == responder {
+		a.responder = getResponder()
+		return &pb.ResetResponse{Msg: success}, nil
+	}
+
+	if in.Service == storage {
+		id := getId(time.Now().String())
+		if err := a.PublishMsg(ctx, id, "reset", nil); err != nil {
+			return nil, err
+		}
+		time.Sleep(time.Millisecond * 15)
+		result, ok := a.responses.Load(id)
+		if !ok {
+			return nil, errors.New(errorMissingResp)
+		}
+		a.responses.Delete(id)
+		return &pb.ResetResponse{Msg: result.(map[string]string)["Value"]}, nil
+	}
+
+	return nil, errors.New(invalidServiceName)
+}
+
+func (a *Responder) GetMode(ctx context.Context, in *pb.GetModeRequest) (*pb.GetModeResponse, error) {
+	if in == nil || in.Service == "" {
+		return nil, errors.New(emptyRequest)
+	}
+
+	if in.Service == "storage" || in.Service == "responder" || in.Service == "portal" {
+		id := getId(time.Now().String())
+		if err := a.PublishMsg(ctx, id, "getMode", in.Service); err != nil {
+			return nil, err
+		}
+		time.Sleep(time.Millisecond * 15)
+		result, ok := a.responses.Load(id)
+		if !ok {
+			return nil, errors.New(errorMissingResp)
+		}
+		a.responses.Delete(id)
+		mode := result.(map[string]string)["Value"]
+
+		if mode == "true" || mode == "false" {
+			return &pb.GetModeResponse{Mode: mode}, nil
+		}
+
+		return nil, errors.New("invalid value from storage")
+	}
+
+	return nil, errors.New(invalidServiceName)
+}
+
+func (a *Responder) SetMode(ctx context.Context, in *pb.SetModeRequest) (*pb.SetModeResponse, error) {
+	if in == nil || in.Service == "" {
+		return nil, errors.New(emptyRequest)
+	}
+
+	if in.Service == storage || in.Service == responder || in.Service == portal {
+		id := getId(time.Now().String())
+		if err := a.PublishMsg(ctx, id, "setMode", in.Service+"&"+strconv.FormatBool(in.Mode)); err != nil {
+			return nil, err
+		}
+		time.Sleep(time.Millisecond * 15)
+		result, ok := a.responses.Load(id)
+		if !ok {
+			return nil, errors.New(errorMissingResp)
+		}
+		a.responses.Delete(id)
+		return &pb.SetModeResponse{Msg: result.(map[string]string)["Value"]}, nil
+	}
+
+	return nil, errors.New(invalidServiceName)
+}
+
+func (a *Responder) EventHandler(ctx context.Context, e *common.TopicEvent) (retry bool, err error) {
+	log.Printf("event - PubsubName: %s, Topic: %s, ID: %s, Data: %s", e.PubsubName, e.Topic, e.ID, e.Data)
```

Comment:
It's better to use loglevels instead of just `Printf`. If this message should be printed only in debug mode, you can use log.Debugf

Reply:
Done, change ```Printf``` on ```logger.Debugf```